dockerclient.go

```diff
@@ -212,6 +212,50 @@ func (client *DockerClient) ContainerChanges(id string) ([]*ContainerChanges, er
 	return changes, nil
 }
 
+func (client *DockerClient) readJSONStream(stream io.ReadCloser, decode func(*json.Decoder) decodingResult) (<-chan decodingResult, chan<- struct{}) {
```

Comment:
1) I can't really reuse or merge with `getEvents` because `getEvents` uses the global `client.monitorEvents` flag, and I really don't want to do that. I can, however, reimplement `StartMonitorEvents` and `StopAllMonitorEvents` using `MonitorEvents` and remove `getEvents` if you prefer that.

The main reason I factored `readJSONStream` out of `MonitorEvents` was so that my stats pr #71 could use it as well
2) okay, that sounds reasonable. I can make that change